### DISCUSSION TITLE
Github Actions

### DISCUSSION THREAD

**Discussion by helio-frota** [nodeshift/new-module-starter#8] [General]
We are moving all the repositories to use Github actions.

But we don't have any ideas on getting this "correct way" or at least plans on how to keep these updated and etc...

**Comment by helio-frota** [comment]
@richardlau any suggestions, hints, comments will be appreciated :+1:

**Comment by helio-frota** [comment]
Since the repo is not private anymore we could probably uncomment these lines https://github.com/nodeshift/new-module-starter/blob/main/.github/workflows/nodejs-ci-action.yml#L27-L48

We have a fooBar test https://github.com/nodeshift/new-module-starter/blob/main/test/test.js#L4
just to make the CI pass and to see the complete flow :+1: 

@richardlau make sense?

**Reply by richardlau** [reply]
Sounds good.

**Comment by helio-frota** [comment]
having this benefit then makes sense for us to start to update all other repositories 
https://github.com/nodeshift/new-module-starter/pull/10

**Comment by helio-frota** [comment]
- [x] new-module-starter https://github.com/nodeshift/new-module-starter/pull/10
- [x] npcheck https://github.com/nodeshift/npcheck/pull/55
- [x] oratrix https://github.com/nodeshift/oratrix/pull/98
- [x] nodejs-cache https://github.com/nodeshift-starters/nodejs-cache/pull/120
- [x] nodejs-rest-http-ts https://github.com/nodeshift-starters/nodejs-rest-http-ts/pull/2
- [x] nodejs-rest-http https://github.com/nodeshift-starters/nodejs-rest-http/pull/210
- [x] license-reporter https://github.com/nodeshift/license-reporter/pull/392
- [x] opossum https://github.com/nodeshift/opossum/pull/595
- [x] rhoaster https://github.com/nodeshift/rhoaster/pull/85
- [x] opossum-prometheus https://github.com/nodeshift/opossum-prometheus/pull/54
- [x] kube-service-bindings https://github.com/nodeshift/kube-service-bindings/pull/8
- [x] nodeshift https://github.com/nodeshift/nodeshift/pull/583
- [x] openshift-rest-client https://github.com/nodeshift/openshift-rest-client/pull/275
- [x] https://github.com/nodeshift-starters/nodejs-health-check/pull/224
- [x] https://github.com/nodeshift-starters/nodejs-configmap/pull/195
- [x] https://github.com/nodeshift-starters/nodejs-rest-http-crud/pull/243

**Comment by helio-frota** [comment]
for the v2 part the missing repositories are:

https://github.com/nodeshift-starters/nodejs-rest-http-secured
https://github.com/nodeshift-starters/reactive-example

the circuit breaker repos are using travis and rhoaster and that is something we can't migrate now to gh actions

- [x] https://github.com/nodeshift-starters/reactive-example/pull/36
- [x] https://github.com/nodeshift-starters/nodejs-rest-http-secured/pull/183

**Comment by helio-frota** [comment]
what happens when we finish this? just lock the discussion or delete it?

**Reply by richardlau** [reply]
I'd rather not delete it. Unfortunately at this time it looks like our options are only delete, lock, or leave open: https://github.com/github/feedback/discussions/3097

**Comment by helio-frota** [comment]
@richardlau thanks for that info.

let's leave it open then
